Automattic/pocket-casts-android
# Setup Google SSO server call

## Description
This change does the following:
- takes the Google id token and sends it to our sync server
- adds logic to handle refresh and access tokens
- updates the forgotten password call to use Retrofit

## Testing Instructions

I recommend using an actual device as I found it more reliable than an emulator with the Google Sign-In framework.
It would be good to test both using the staging and production servers as they are running different code and we will be releasing the Android app before the updated server code is released to production.

### Staging testing setup
1. Change SINGLE_SIGN_ON_ENABLED in base.gradle to `true`
2. Make sure the Build Variant is 'debug'

### Test Google sign in, with an existing account
1. Clean install the app
2. Tap 'Continue with Google'
3. Pick your Google account with an email that exists in the app
4. ✅ Verify you are taken into the app and on the profile tab your email should be displayed

### Test Google sign in, without an existing account
1. Clean install the app
2. Tap 'Continue with Google'
3. Pick your Google account with an email that doesn't exist (if you need delete the account in the staging user admin area)
4. ✅ Verify you are taken to the next welcome page to pick some podcasts

### Test email sign in
1. Clean install the app
2. Tap 'Sign up'
3. Enter a new email and password
4. Tap 'Create Account'
5. ✅ Verify you are taken to the next welcome page to pick some podcasts

### Test creating an account with email
1. Clean install the app
2. Tap 'Sign up'
3. Enter a new email and password
4. Tap 'Create Account'
5. ✅ Verify you are taken to the next welcome page to pick some podcasts

### Test forgotten password
1. Clean install the app
2. Tap 'Log in'
3. Enter an existing username and password
4. Tap 'Log in'
5. ✅ Verify you are taken to the next welcome page to pick some podcasts

### Production testing setup
1. Change SINGLE_SIGN_ON_ENABLED in base.gradle to `true`
2. Make sure the Build Variant is 'debugProd'
3. Repeat the tests above, excluding the Google sign in tests, as the server endpoints aren't available yet.
